src/wallet/coinselection.cpp

```diff
@@ -0,0 +1,370 @@
+#include "../tweak.h"
+#include "random.h"
+#include "wallet.h"
+
+// needed for tx validation
+#include "../main.h"
+#include "../script/sign.h"
+#include "../txmempool.h"
+const int MAX_SOLUTIONS = 2000; // The approximate maximum number of solutions we will find before giving up.
+// What's the maximum number of TXOs we should use as inputs to a transaction (if we have a choice).
+const int MAX_ELECTIVE_TXOS = 5;
+const int MAX_TXOS = 100; // What's the maximum number of TXOs to put in a transaction
+// how many satoshi's the coin selection is willing to be off by per iteration through the loop.  Makes it progressively
+// easier to find a solution
+// (now its a simple exponential) const long int LOOP_COST = 500;
+const int P2PKH_INPUT_SIZE = 72; // <Sig> <PubKey> OP_DUP OP_HASH160 <PubkeyHash> OP_EQUALVERIFY OP_CHECKSIG
+
+extern CTweak<unsigned int> maxCoinSelSearchTime;
+extern CTweak<unsigned int> preferredNumUTXO;
+
+#if 0 // for reference. Defined in wallet.h
+typedef std::multimap<CAmount, COutput> SpendableTxos;
```

Comment:
do you still need this reference which is defined in wallet.h?

Reply:
removed